src/lib/vault/encryptFormSubmission.ts

```diff
@@ -0,0 +1,40 @@
+import { getPublicKey } from "./getPublicKey.js";
+import type { FormSubmission } from "./dataStructures/formSubmission.js";
+import {
+  randomBytes,
+  createCipheriv,
+  createPublicKey,
+  publicEncrypt,
+} from "node:crypto";
+
+export const encryptFormSubmission = async (
+  serviceAccountId: string,
+  submission: FormSubmission,
+) => {
+  const serviceAccountPublicKey = await getPublicKey(serviceAccountId);
+
+  // Encrypt the submission with the public key
+  const encryptionKey = randomBytes(32);
+  const iv = randomBytes(16);
+
+  const cipher = createCipheriv("aes-256-gcm", encryptionKey, iv);
+
+  const encryptedResponses = Buffer.concat([
+    cipher.update(Buffer.from(JSON.stringify(submission))),
+    cipher.final(),
+  ]);
+  const authTag = cipher.getAuthTag();
+  const publicKey = createPublicKey({ key: serviceAccountPublicKey });
+  const encryptedKey = publicEncrypt(publicKey, encryptionKey).toString(
+    "base64",
+  );
```

Comment:
I was reading that we could improve the strength of our encryption with a change to the hashing algorithm and padding used:
```suggestion
  const encryptedKey = publicEncrypt({
    key: publicKey,
    padding: crypto.constants.RSA_PKCS1_OAEP_PADDING,
    oaepHash: "sha256",  // default of the function is `sha1`
   }, encryptionKey).toString("base64");
```
Above is untested snippet from the internet.

Reply:
We can definitely try it out.  The oaepHash with the `sha1` default is only being used as a random generator and not as an actual hashing function in this case because we're not passing in an `oaepLabel`.  We'll also need to checkout other programming languages crypto modules too see what the commonalities are across them and what are the most prevalent settings.  
What do you thing about flagging it in an issue and revisiting it once we start looking at building an example folder with other programming languages?

Reply:
Totally fine with flagging and addressing once we've had a chance to look at other programming languages.